libraries/AP_ICEngine/AP_ICEngine.cpp

```diff
@@ -343,6 +378,20 @@ bool AP_ICEngine::throttle_override(uint8_t &percentage)
         percentage = (uint8_t)start_percent.get();
         return true;
     }
+
+    if (redline_flag && !(options & uint16_t(Options::DISABLE_REDLINE_GOVERNOR))) {
+        // limit the throttle from increasing above what the current output is
+        if (redline_throttle_percentage < 1.0f) {
+            redline_throttle_percentage = SRV_Channels::get_output_scaled(SRV_Channel::k_throttle);
+        }
+        if (filtered_rpm_value > redline_rpm) {
+            // reduce the throttle if still over the redline RPM
+            static const float redline_setpoint_step = idle_slew * AP::scheduler().get_loop_period_s();
```

Comment:
does this need to be static?

Reply:
Monkey see monkey do from the idle governor. Probably not now that I'm reading stackoverflow.